src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/Package/Update/PackageUpdateIO.cs

```diff
@@ -318,12 +320,36 @@ public async Task<IReadOnlyList<IReadOnlyDictionary<string, IReadOnlyList<Packag
         return lowestNonVulnerableVersion;
     }
 
-    private List<SourceRepository> GetSourcesForPackage(string packageId)
+    public PackageSourceMapping GetPackageSourceMapping()
     {
-        var sources = new List<SourceRepository>(_enabledSources.Count);
-        for (int i = 0; i < _enabledSources.Count; i++)
+        return PackageSourceMapping.GetPackageSourceMapping(_settings);
```

Comment:
question: Should this be cached? 
I don't tihnk it's called more than once in the current implementation but from an API perspective I wonder if it needs to be, since the ISettings object at this point is assumed to no longer be mutable.